contracts/DataFeeds/TimeMedianDataFeed.sol

```diff
@@ -0,0 +1,145 @@
+pragma solidity ^0.4.24;
+
+import "tidbit/contracts/DataFeedOracles/DataFeedOracleBase.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract TimeMedianDataFeed is DataFeedOracleBase {
+  using SafeMath for uint;
+
+  mapping(uint => bool) dateAlreadyAccountedFor; // transitory data structure useful only during function call medianizeByDates
+
+  function medianizeByTimeframe(uint startDate, uint endDate)
+    public
+    view
+    returns (bytes32 medianValue)
+  {
+    require(startDate < endDate, 'startDate must be less than endDate');
+    uint endIndex;
+    uint startIndex;
+
+    // find end date within dates array
+    for (uint i = dates.length - 1; i > 0 && endIndex == 0; i--) {
+      if (dates[i] <= endDate) {
+        endIndex = i; // ONLY dates before the end date
+      }
+    }
+
+    // require endIndex is more than 0
+    require(endIndex > 0, 'dates outside of range of datafeed results');
+    require(dates[endIndex] >= startDate, 'dates outside of range of datafeed results');
+
+    // find start date within dates array
+    for (uint j = endIndex; j >= 0 && startIndex == 0; j--) {
+      if (dates[j] < startDate) {
+        startIndex = j + 1; // no dates before the startDate
+      }
+    }
+
+    medianValue = medianizeByIndices(startIndex, endIndex);
+  }
+
+  function medianizeByIndices(uint startIndex, uint endIndex)
+    public
+    view
+    returns (bytes32 medianValue)
+  {
+    require(startIndex <= endIndex, 'startIndex must be less than or equal to endIndex');
+    require(startIndex > 0, 'startIndex must be greater than 0');
```

Comment:
should this be `>= 0`?

Reply:
in tidbit, index 1 will be the first index with an assignment. index 0 is reserved as nil

Reply:
coool